prow/cmd/peribolos/main_test.go

```diff
@@ -44,23 +50,77 @@ func TestOptions(t *testing.T) {
 			name: "bad --github-endpoint",
 			args: []string{"--config-path=foo", "--github-token-path=bar", "--github-endpoint=ht!tp://:dumb"},
 		},
+		{
+			name: "--minAdmins too low",
+			args: []string{"--config-path=foo", "--github-token-path=bar", "--min-admins=1"},
```

Comment:
It's not clear to me what the utility is of checking that Go's `"flag"` stdlib works, why not just test from the `Options` struct forward?

Reply:
Because `--config-path=foo` is user input that we need to validate does not regress. 

On the other hand `options{ config: "foo"}` is an implementation detail we can change without worrying about breaking anyone.

Testing the effect of different `maximumDelta` values happens in `TestConfigureOrgMembers` and does use the `options` struct.

This does not validate the flag package. It does not, for example, test that it handles `--config-path foo` and `--config-path=foo` correctly. Nor does it ensure it test the behavior of replacing `foo` with really long, short, non-english characters. Nor the behavior of non-recognized flags, etc.

It ensures that if someone changes this:
`flags.Var(&o.requiredAdmins, "required-admins", "Ensure config specifies these users as admins")`
to:
`flags.Var(&o.requiredAdmins, "required-admin", "Ensure config specifies these users as admins")`
that a unit test will fail and cause them to hopefully think about whether or not they should really be doing that.

Reply:
But if I were to refactor your code to use `cobra` instead of `flag` and keep the flag identically named, we'd still fail. So this is kind of an integration test but at the same time it still doesn't black-box the flag parsing and depends on an implementation detail of the specific flag parsing library used. Do we have tests like this in `k/k`? IIRC the majority of tests for user interaction on the CLI are done in shell as that's how users will actually interact with the binary.